src/utils/DatePick.tsx

```diff
@@ -0,0 +1,71 @@
+import DatePicker from 'react-datepicker'
+import 'react-datepicker/dist/react-datepicker.css'
+import { ko } from 'date-fns/locale'
+import inputCalendar from '@/images/inputCalendar.svg'
+import { forwardRef, useState } from 'react'
+import { format } from 'date-fns'
+
+interface DatePickProps {
+	selectedDate: Date
+	minDate?: Date
+	maxDate?: Date
+	textSize?: string
+	handleSelect: (date: Date) => void
+}
+
+interface DateInputProps {
+	onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+export const DatePick = ({
+	selectedDate,
+	minDate = new Date(0, 0),
+	maxDate = new Date(),
+	handleSelect,
+	textSize = 'text-md',
+}: DatePickProps) => {
+	const [open, setOpen] = useState(false)
+
+	const DateInput = forwardRef<HTMLButtonElement, DateInputProps>(
+		({ onClick }, ref) => {
+			const formattedDate = selectedDate
+				? format(selectedDate, 'yyyy-MM-dd')
+				: ''
+
+			return (
+				<button
+					className='example-custom-input hover:bg-gray-200 hover:duration-300'
+					onClick={onClick}
+					ref={ref}
+				>
+					{formattedDate}
+				</button>
+			)
+		},
+	)
+
+	return (
+		<DatePicker
+			icon={<img src={inputCalendar} />}
+			className={`${textSize}`}
+			// 기본 Input창의 패딩이 너무 큰데, 이걸 수정할 수가 없어서 커스텀 인풋을 만듬
+			customInput={<DateInput onClick={() => setOpen(true)} />}
+			locale={ko}
+			dateFormat='yyyy-MM-dd'
+			//state를 넘길지 말지 고민 많이 했는데 넘기는게 맞는듯, 이거 아니면 오히려 복잡한데 결국 상태를 받아야함
+			selected={selectedDate}
+			open={open}
+			minDate={minDate}
+			maxDate={maxDate}
+			onInputClick={() => setOpen(true)}
+			onClickOutside={() => setOpen(false)}
+			onSelect={(date) => {
```

Comment:
현재 date에 타입 정의가 되어 있지 않은 것 같습니다. 타입 정의를 해주는 편이 어떨까요?
```suggestion
			onSelect={(date: Date) => {
```
`onSelect` 로직은 나중에 예외 처리 및 validation을 할 때 `handleDateSelect`로 분리해주는 것이 좋을 것 같습니다.

Reply:
onSelect가 라이브러리에서 자체적으로 제공하는 함수라 타입이 따로 있어서요 Date | null로 자동 추론이 됩니다.

Reply:
handleSelect 타입정의를 보니 Date타입만 허용하고 있어 코멘트 달았습니다 ㅎㅎ 무조건 Date만 받도록 하시는 걸로 이해하고 null처리문도 불필요하다고 생각했네요.